k8s/production/custom/gitlab-pod-cleanup/pod.yaml

```diff
@@ -0,0 +1,49 @@
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: pod-cleanup-role
+rules:
+  - apiGroups: [""]
+    resources: ["pods"]
+    verbs: ["get", "list", "delete"]
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: pod-cleanup-role-binding
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: pod-cleanup-role
+subjects:
+  - kind: ServiceAccount
+    name: pod-cleanup-sa
+    namespace: custom
+---
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: pod-cleanup-sa
+  namespace: custom
+---
+apiVersion: v1
+kind: Pod
+metadata:
+  name: gitlab-runner-pod-cleanup
+  namespace: custom
```

Comment:
Should we have this in the `gitlab` namespace instead? It seems like we generally reserve `custom` for custom scripts/jobs, whereas this is a tool maintained by gitlab.

Reply:
I was thinking something similar as well. I'll change it.